pools-latency-calculator/monitor_and_apply_latency.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+# Verifica se sono stati forniti l'IP di destinazione e il divisore come argomenti
+if [ -z "$1" ] || [ -z "$2" ]; then
+    echo "Usage: $0 <TARGET_IP> <DIVISOR>"
+    exit 1
+fi
+
+TARGET_IP="$1"
+DIVISOR="$2"
+
+# URL of the Prometheus query
+PROMETHEUS_QUERY_URL="http://10.5.0.9:9090/api/v1/query?query=average_pool_subscription_latency_milliseconds"
+
+while true; do
+    # Get the latency value from the Prometheus query
+    JSON=$(curl -s $PROMETHEUS_QUERY_URL)
+    LATENCY=$(echo "$JSON" | grep -o '"value":[[0-9.]*,"[0-9.]*' | cut -d',' -f2 | tr -d '"')
+
+    # Check if LATENCY is empty
+    if [ -z "$LATENCY" ]; then
+        echo "No latency value found, skipping this cycle."
+    else
+        # Set the latency using the tc command, dividing the value obtained from the query by the divisor
+        ADJUSTED_LATENCY=$(echo "$LATENCY" | awk -v div="$DIVISOR" '{print $1 / div}')
```

Comment:
Can we go for an approach that involves maintaining a history of latency values and dynamically calculating the 95th percentile (p95) latency, rather than using a fixed divisor for latency adjustment?

Reply:
`tc` already introduces a dynamic error (+-10%) according to its docs: https://man7.org/linux/man-pages/man8/tc-netem.8.html#EXAMPLES
The divisor is just because for the SV1 context, the latency imposed needs to be halved, since it's applied twice for every round-trip. All the others are just fine as they are.